src/Forms/Shared/Samples/Location/IndoorPositioning/IndoorPositioning.xaml.cs

```diff
@@ -0,0 +1,204 @@
+        instructions: "Bring the device within range of an IPS beacon. The system will ask for permission to use the device's location if the user has not yet used location services in this app. It will then start the location display with auto-pan mode set to `Navigation`.",
+        tags: new[] { "BLE", "Bluetooth", "GPS", "IPS", "beacon", "blue dot", "building", "facility", "indoor", "location", "map", "mobile", "navigation", "site", "transmitter" })]
+    public partial class IndoorPositioning : ContentPage, IDisposable
+    {
+        private IndoorsLocationDataSource _indoorsLocationDataSource;
+
+        private int? _currentFloor = null;
+
+        // This data is specific to a building on the Esri campus. Substitute your own data in order to use this sample. Code in the sample may need to be modified to work with other maps.
+
+        #region EsriBuildingData
+
+        private Uri _portalUri = new Uri("https://viennardc.maps.arcgis.com");
+
+        private const string sampleUser = "tester_viennardc";
+        private const string samplePass = "password.testing12345";
+
+        private const string ItemId = "89f88764c29b48218366855d7717d266";
+
+        private const string PositioningTableName = "ips_positioning";
+        private const string PathwaysLayerName = "pathways";
+
+        private string[] _layerNames = new string[] { "Details", "Units", "Levels" };
+
+        #endregion EsriBuildingData
+
+        public IndoorPositioning()
+        {
+            InitializeComponent();
+            _ = Initialize();
+        }
+        private async Task Initialize()
+        {
+            // Handle the login to the feature service.
+            AuthenticationManager.Current.ChallengeHandler = new ChallengeHandler(async (info) =>
+            {
+                try
+                {
+                    // WARNING: Never hardcode login information in a production application. This is done solely for the sake of the sample.
+                    return await AuthenticationManager.Current.GenerateCredentialAsync(info.ServiceUri, sampleUser, samplePass, info.GenerateTokenOptions);
+                }
+                catch (Exception ex)
+                {
+                    Debug.WriteLine(ex.Message);
+                    return null;
+                }
+            });
+
+            try
+            {
+                try
+                {
+                    Xamarin.Essentials.PermissionStatus status = await Xamarin.Essentials.Permissions.RequestAsync<Xamarin.Essentials.Permissions.LocationWhenInUse>();
+                    if (status != Xamarin.Essentials.PermissionStatus.Granted)
+                    {
+                        await Application.Current.MainPage.DisplayAlert("Error", "Location permission required for use of indoor positioning.", "OK");
+                        return;
+                    }
+                }
+                catch
+                {
+                    return;
+                }
```

Comment:
Instead of try/catch and displaying alert, you could just throw, and the display alert code further down will handle it - this would reduce code duplication.